IntegrationAdapters/Controllers/MedicineController.cs

```diff
@@ -1,35 +1,78 @@
 ﻿using IntegrationAdapters.Models;
+using IntegrationAdapters.Models.DTO;
 using IntegrationAdapters.Services;
 using Microsoft.AspNetCore.Mvc;
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
-using System.Threading.Tasks;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
 
 namespace IntegrationAdapters.Controllers
 {
     [Route("[controller]")]
     [ApiController]
-    public class MedicineController
+    public class MedicineController : ControllerBase
     {
         private readonly MedicineService medicineService;
 
         public MedicineController()
         {
             this.medicineService = new MedicineService();
         }
-        [HttpGet("getMedicineSpecification/{MedicineName}")]
-        public String GetMedicineSpeification(String MedicineName)
+        [HttpGet("getMedicineSpecification/{medicineName}")]
+        public IActionResult GetMedicineSpeification(String medicineName)
         {
-            Medicine medicine = medicineService.GetMedicineByName(MedicineName);
-            if (medicineService.GetMedicineByName(MedicineName) != null)
+            Medicine medicine = medicineService.GetMedicineByName(medicineName);
+            if (medicineService.GetMedicineByName(medicineName) != null)
             {
-                return medicine.MedicineSpecificationID;
+                MedicineSpecification medicineSpecification = medicineService.GetById(medicine.MedicineSpecificationID);
+                medicineService.GenerateSpecificationFromHospital(medicineName, medicineSpecification);
```

Comment:
I would join there two methods, medicineService.GenerateSpecificationFromHospital(medicineName) - that method should get the medicine by name, get the medicine specification and then generate the file - this could be divided into some private methods